lizard_auth_server/views.py

```diff
@@ -304,3 +324,190 @@ def profile(self):
         if not self._profile:
             self._profile = self.invitation.user.get_profile()
         return self._profile
+
+
+class JWTView(FormView):
+    """A view that provides a JSON Web Token after successful authentication.
+
+    The query string may have a `next` parameter used for redirection. If
+    present, it must be a fully qualified URL starting with http(s). The
+    token will be attached to the query string as an extra parameter:
+    <next>?foo=bar&access_token=xxxxx.yyyyy.zzzzz&...
+
+    If a `next` parameter is absent, a JSON response is returned:
+
+    {"access_token": "xxxxx.yyyyy.zzzzz"}
+
+    """
+    template_name = 'lizard_auth_server/login.html'
+    form_class = forms.JWTAuthenticationForm
+
+    @staticmethod
+    def is_url(url):
+        """Validate a URL.
+
+        The current validator accepts only absolute URLs, which makes
+        sense as the token is to be generated for another portal.
+
+        Args:
+            url (str): an absolute, i.e. fully qualified, URL.
+
+        Returns:
+            bool: True if valid, False otherwise.
+
+        """
+        val = URLValidator()
+        try:
+            val(url)
+            return True
+        except ValidationError:
+            return False
+
+    @staticmethod
+    def is_portal(sso_key):
+        """Validate a portal.
+
+        Args:
+            sso_key (str): a public key identifying.
+
+        Returns:
+            bool: True if the portal exists, False otherwise.
+
+        """
+        return Portal.objects.filter(sso_key=sso_key).exists()
+
+    def get(self, request, *args, **kwargs):
+        """Return login form.
+
+        The user already is authenticated. Do not present a login form,
+        but give him what he wants right away. This can be used to
+        easily refresh expired tokens.
+
+        """
+        redirect_to = request.GET.get('next', '')
+        sso_key = request.GET.get('portal', '')
+
+        if redirect_to and not JWTView.is_url(redirect_to):
+            reason = _('Invalid `next` query string parameter.')
+            return HttpResponseBadRequest(reason, content_type='text/plain')
+
+        if not sso_key:
+            reason = _('Missing `portal` query string parameter.')
+            return HttpResponseBadRequest(reason, content_type='text/plain')
+
+        if not JWTView.is_portal(sso_key):
+            reason = _('Invalid `portal` query string parameter.')
+            return HttpResponseBadRequest(reason, content_type='text/plain')
+
+        self.redirect_to = redirect_to
+        self.sso_key = sso_key
+
+        if not (hasattr(request, 'user') and request.user.is_authenticated()):
```

Comment:
Is er een manier waarom het request geen user attribuut heeft? (Indien het alleen in tests is, misschien de test aanpassen, want normaliter is gewoon kaal `request.user.is_authenticated()` genoeg.